FXyz-Importers/src/main/java/org/fxyz3d/importers/obj/MtlReader.java

```diff
@@ -54,87 +53,143 @@
-        BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
-        String line;
-        String name = "default";
-        while ((line = br.readLine()) != null) {
-            try {
-                if (line.isEmpty() || line.startsWith("#")) {
-                    // comments and empty lines are ignored
-                } else if (line.startsWith("newmtl ")) {
-                    addMaterial(name);
-                    name = line.substring("newmtl ".length());
-                } else if (line.startsWith("Kd ")) {
-                    material.setDiffuseColor(readColor(line.substring(3)));
-                    modified = true;
-                } else if (line.startsWith("Ks ")) {
-                    material.setSpecularColor(readColor(line.substring(3)));
-                    modified = true;
-                } else if (line.startsWith("Ns ")) {
-                    material.setSpecularPower(Double.parseDouble(line.substring(3)));
-                    modified = true;
-                } else if (line.startsWith("map_Kd ")) {
-                    material.setDiffuseColor(Color.WHITE);
-                    material.setDiffuseMap(loadImage(line.substring("map_Kd ".length())));
-//                    material.setSelfIlluminationMap(loadImage(line.substring("map_Kd ".length())));
-//                    material.setSpecularColor(Color.WHITE);
-                    modified = true;
-                    //            } else if (line.startsWith("illum ")) {
-                    //                int illumNo = Integer.parseInt(line.substring("illum ".length()));
-                    /*
-                        0    Color on and Ambient off
-                        1    Color on and Ambient on
-                        2    Highlight on
-                        3    Reflection on and Ray trace on
-                        4    Transparency: Glass on
-                             Reflection: Ray trace on
-                        5    Reflection: Fresnel on and Ray trace on
-                        6    Transparency: Refraction on
-                             Reflection: Fresnel off and Ray trace on
-                        7    Transparency: Refraction on
-                             Reflection: Fresnel on and Ray trace on
-                        8    Reflection on and Ray trace off
-                        9    Transparency: Glass on
-                             Reflection: Ray trace off
-                        10   Casts shadows onto invisible surfaces
-                     */
-                } else {
-                    //log("material line ignored for " + name + ": " + line);
-                }
-            } catch (Exception ex) {
-                Logger.getLogger(MtlReader.class.getName()).log(Level.SEVERE, "Failed to parse line:" + line, ex);
+    private PhongMaterial currentMaterial;
+
+    // mtl format spec: http://paulbourke.net/dataformats/mtl/
+    private enum ParseKey {
```

Comment:
The enum looks ok, but why do we have two different approaches:  obj importer uses the Map with the predicates and biconsumers, mtl importer uses enums... shouldn't we use the same (I don't say which one of them)?

Reply:
We should. I wanted to consult to see if there are opinions towards one or the other. Honestly, even an `if-else` chain isn't *that* bad.
Part of the change is to remove the `line.substring(n)` job from the parsing function, where `n` is a magic number equal to the length of the key. I think I'm going back to the `Map` approach now that I thought about it a bit more and realized I can make it work with minor changes.